lib/cext/preprocess.rb

```diff
@@ -305,6 +305,66 @@ def read_write_array(name, leaking)
         replacement: BYEBUG_THREADS_INIT
       }
     ]
+  },
+  'parser.c' => {
+    gem: 'json',
+    ext_dir: 'parser',
+    patches: [
+      *read_write_field('json','Vsource', false),
+      *read_write_field('json','create_id', false),
+      *read_write_field('json','object_class', false),
+      *read_write_field('json','array_class', false),
+      *read_write_field('json','decimal_class', false),
+      *read_write_field('json','match_string', false),
+      { # cParser_initialize
+        match: /if \(rb_tr_managed_from_handle\(json->Vsource\)\)/,
+        replacement: 'if (rb_tr_managed_from_handle_or_null(json->Vsource))'
+      },
+      { # cParser_parse
+        match: /VALUE result = Qnil;/,
+        replacement: "VALUE result[1];\nresult[0] = Qnil;"
+      },
+      { # cParser_parse
+        match: /char \*np = JSON_parse_value\(json, p, pe, &result, 0\);/,
+        replacement: 'char *np = JSON_parse_value(json, p, pe, result, 0);'
+      },
+      {
+        match: /if \(cs >= JSON_first_final && p == pe\) {\s+return result;/m,
+        replacement: "if (cs >= JSON_first_final && p == pe) {\n return result[0];"
+      },
```

Comment:
It sounds better to renamed the variable here, so a missed `&result` in the code would fail compilation and not silently do the wrong thing.

Reply:
Fair enough. Although I could see it making debugging more complicated since we don't materialize the preprocess changes to disk.

Reply:
It could follow some convention like `<var>_array_for_address` or so.